File: server/models/document.py
"""
文档模型
"""
from datetime import datetime
from database import db


class Document(db.Model):
    """文档模型"""
    __tablename__ = 'documents'
    
    id = db.Column(db.Integer, primary_key=True)
    meeting_id = db.Column(db.String(36), db.ForeignKey('meetings.id'), nullable=False, index=True)
    filename = db.Column(db.String(255), nullable=False)
    original_filename = db.Column(db.String(255), nullable=False)  # 原始文件名
    file_path = db.Column(db.String(500), nullable=False)  # 文件存储路径
    file_size = db.Column(db.Integer, nullable=False)  # 文件大小（字节）
    file_type = db.Column(db.String(50), nullable=False)  # 文件类型：pdf, docx, pptx, txt等
    mime_type = db.Column(db.String(100), nullable=True)  # MIME类型
    status = db.Column(db.String(20), default='uploaded', nullable=False)  # uploaded, processing, completed, failed
    parsed_content = db.Column(db.Text, nullable=True)  # 解析后的文本内容
    parse_progress = db.Column(db.Integer, default=0, nullable=False)  # 解析进度 0-100
    error_message = db.Column(db.Text, nullable=True)  # 错误信息
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False, index=True)
    created_at = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
    
    def to_dict(self):
        """转换为字典"""
        return {
            'id': self.id,
            'meeting_id': self.meeting_id,
            'filename': self.filename,
            'original_filename': self.original_filename,
            'file_path': self.file_path,
            'file_size': self.file_size,
            'file_size_mb': round(self.file_size / (1024 * 1024), 2),
            'file_type': self.file_type,
            'mime_type': self.mime_type,
            'status': self.status,
            'parse_progress': self.parse_progress,
            'error_message': self.error_message,
            'user_id': self.user_id,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None,
        }
    
    def __repr__(self):
        return f'<Document {self.original_filename}>'

File: frontend/src/services/document.ts
/**
 * 文档服务 - 处理文档上传相关的API调用
 */

const API_BASE_URL = ''

export interface Document {
  id: number
  meeting_id: string
  filename: string
  original_filename: string
  file_path: string
  file_size: number
  file_size_mb: number
  file_type: string
  mime_type: string
  status: 'uploaded' | 'processing' | 'completed' | 'failed'
  parse_progress: number
  error_message?: string
  user_id: number
  created_at: string
  updated_at: string
}

export interface DocumentListResponse {
  success: boolean
  data: Document[]
  total: number
}

export interface DocumentResponse {
  success: boolean
  data: Document
}

/**
 * 获取请求头（包含认证token）
 */
function getHeaders(): HeadersInit {
  const headers: HeadersInit = {}

  const token = localStorage.getItem('access_token')
  if (token) {
    headers['Authorization'] = `Bearer ${token}`
  }

  return headers
}

/**
 * 上传文档
 */
export async function uploadDocument(
  meetingId: string,
  file: File,
): Promise<Document> {
  const formData = new FormData()
  formData.append('file', file)

  const response = await fetch(`${API_BASE_URL}/api/documents/upload/${meetingId}`, {
    method: 'POST',
    headers: getHeaders(),
    body: formData,
  })

  const data: DocumentResponse = await response.json()

  if (!response.ok) {
    throw new Error(data.success === false ? data.message || '上传文档失败' : '未授权')
  }

  return data.data
}

/**
 * 获取会议的所有文档
 */
export async function getDocuments(meetingId: string): Promise<Document[]> {
  const response = await fetch(`${API_BASE_URL}/api/documents/list/${meetingId}`, {
    method: 'GET',
    headers: getHeaders(),
  })

  const data: DocumentListResponse = await response.json()

  if (!response.ok) {
    throw new Error(data.success === false ? data.message || '获取文档列表失败' : '未授权')
  }

  return data.data || []
}

/**
 * 删除文档
 */
export async function deleteDocument(documentId: number): Promise<void> {
  const response = await fetch(`${API_BASE_URL}/api/documents/${documentId}`, {
    method: 'DELETE',
    headers: getHeaders(),
  })

  const data = await response.json()

  if (!response.ok) {
    throw new Error(data.success === false ? data.message || '删除文档失败' : '未授权')
  }
}

/**
 * 获取文件类型图标
 */
export function getFileTypeIcon(fileType: string): string {
  const iconMap: Record<string, string> = {
    pdf: 'PDF',
    doc: 'DOC',
    docx: 'DOC',
    ppt: 'PPT',
    pptx: 'PPT',
    txt: 'TXT',
    md: 'TXT',
    xls: 'XLS',
    xlsx: 'XLS',
  }
  return iconMap[fileType.toLowerCase()] || 'FILE'
}

/**
 * 获取文件类型颜色类
 */
export function getFileTypeColorClass(fileType: string): string {
  const colorMap: Record<string, string> = {
    pdf: 'bg-red-100 text-red-600',
    doc: 'bg-blue-100 text-blue-600',
    docx: 'bg-blue-100 text-blue-600',
    ppt: 'bg-orange-100 text-orange-600',
    pptx: 'bg-orange-100 text-orange-600',
    txt: 'bg-gray-100 text-gray-600',
    md: 'bg-gray-100 text-gray-600',
    xls: 'bg-green-100 text-green-600',
    xlsx: 'bg-green-100 text-green-600',
  }
  return colorMap[fileType.toLowerCase()] || 'bg-gray-100 text-gray-600'
}

